imap_processing/pointing_frame_handler.py

```diff
@@ -0,0 +1,242 @@
+"""
+Generate Pointing Frame.
+
+Notes
+-----
+Kernels that are required to run this code:
+1. imap_science_0001.tf - pointing frame kernel
+2. imap_sclk_0000.tsc - spacecraft clock kernel
+3. imap_wkcp.tf - spacecraft frame kernel
+4. de430.bsp - standard SPICE planetary ephemeris kernel
+5. naif0012.tls - standard NAIF leapsecond kernel
+6. imap_spin.bc - test attitude kernel available at:
+   https://lasp.colorado.edu/galaxy/display/IMAP/Data
+These need to be placed in tests/pointing_frame/test_data.
+
+References
+----------
+https://spiceypy.readthedocs.io/en/main/documentation.html.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import numpy as np
+import spiceypy as spice
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# TODO : Add multiple pointings to the pointing frame.
+
+
+def get_coverage(ck_kernel: str) -> tuple[float, float, np.ndarray]:
+    """
+    Create the pointing frame.
+
+    Parameters
+    ----------
+    ck_kernel : str
+        Path of ck_kernel used to create the pointing frame.
+
+    Returns
+    -------
+    et_start : float
+        Start time of ck_kernel.
+    et_end : float
+        End time of ck_kernel.
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+    """
+    # Each spin is 15 seconds. We want 10 quaternions per spin.
+    # duration / # samples (nominally 15/10 = 1.5 seconds)
+    step = 1.5
+
+    # Get the spacecraft ID.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+    id_imap_spacecraft = spice.gipool("FRAME_IMAP_SPACECRAFT", 0, 1)
+
+    # TODO: Queried pointing start and stop times here.
+
+    # Get the coverage window
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckcov
+    cover = spice.ckcov(ck_kernel, int(id_imap_spacecraft), True, "SEGMENT", 0, "TDB")
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.wnfetd
+    et_start, et_end = spice.wnfetd(cover, 0)
```

Comment:
As I understand it there are sometimes segments within the kernel so the start and end time would be for each segment. That isn't the case here since we generated the test data. Also this will all go away when we use the et_start and et_end of the queried csv for pointing.